firetv/__init__.py

```diff
@@ -116,57 +116,215 @@ def connect(self):
+                self.adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host, rsa_keys=[signer])
             else:
-                self._adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host)
+                self.adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host)
         except socket_error as serr:
             logging.warning("Couldn't connect to host: %s, error: %s", self.host, serr.strerror)
 
+    def app_state(self, app):
+        """ Informs if application is running """
+        if not self.adb or not self.screen_on:
+            return STATE_OFF
+        if self.current_app["package"] == app:
+            return STATE_ON
+        return STATE_OFF
+
+    def launch_app(self, app):
+        if not self.adb:
+            return None
+
+        return self._send_intent(app, INTENT_LAUNCH)
+
+    def stop_app(self, app):
+        if not self.adb:
+            return None
+
+        return self._send_intent(PACKAGE_LAUNCHER, INTENT_HOME)
+
+    # ======================================================================= #
+    #                                                                         #
+    #                               properties                                #
+    #                                                                         #
+    # ======================================================================= #
     @property
     def state(self):
         """ Compute and return the device state.
 
         :returns: Device state.
         """
         # Check if device is disconnected.
-        if not self._adb:
+        if not self.adb:
             return STATE_UNKNOWN
         # Check if device is off.
-        if not self._screen_on:
+        if not self.screen_on:
             return STATE_OFF
         # Check if screen saver is on.
-        if not self._awake:
+        if not self.awake:
             return STATE_IDLE
         # Check if the launcher is active.
-        if self._launcher or self._settings:
+        if self.launcher or self.settings:
             return STATE_STANDBY
         # Check for a wake lock (device is playing).
-        if self._wake_lock:
+        if self.wake_lock:
             return STATE_PLAYING
         # Otherwise, device is paused.
         return STATE_PAUSED
 
+    @property
     def running_apps(self):
         """ Return an array of running user applications """
         return self._ps('u0_a')
 
-    def app_state(self, app):
-        """ Informs if application is running """
-        if not self._adb or not self._screen_on:
-            return STATE_OFF
-        if self.current_app["package"] == app:
-            return STATE_ON
-        return STATE_OFF
+    @property
+    def current_app(self):
+        current_focus = self._dump("window windows", "mCurrentFocus").replace("\r", "")
+
+        #logging.error("Current: %s", current_focus)
+        #mCurrentFocus = Window{299091cd u0 com.netflix.ninja / com.netflix.ninja.MainActivity}
```

Comment:
Please remove debug, or convert to `logging.debug`

Reply:
Removed in https://github.com/JeffLIrion/python-firetv/commit/e0e0c24933df74afc7780eda900ce0c75971d1de